JS1 Homework/Week2

Shvan1 opened the pull request.

Shvan1 committed:
Add files via upload

marcianoviereck commented on app.js:
```diff
@@ -0,0 +1,77 @@
+
+console.log("I'm awesome!");
+
+let x =
```
I don't think this will compile:) 

I think you wanted to do this?
`let x;` ?

Shvan1 replied:
let x;

Shvan1 replied:
It is solved

marcianoviereck commented on app.js:
```diff
@@ -0,0 +1,77 @@
+console.log("I'm awesome!");
+
+let x =
+console.log('I think the value of x is bigger than 5');
```
This is also not correct, because the value of `x` is not set (yet)

Shvan1 replied:
changed to : "I think the value of x is undefined"

marcianoviereck commented on app.js:
```diff
@@ -0,0 +1,77 @@
+console.log('I think the value of x is bigger than 5');
+console.log("The value of x is 6");
+let x = 6;
+console.log(I think the value of x is 6);
```
I think that this will will give an error message.


Did you mean to do this:

`console.log("I think the value of x is 6");`
?

Shvan1 replied:
Changed to: console.log("the value of x: 6");

marcianoviereck commented on app.js:
```diff
@@ -0,0 +1,77 @@
+var round =Math.round(7.25);
+console.log(round)
+
+let arrays =
```
Same here as above, I don't think this will compile

Shvan1 replied:
Changed to : let arrays;

marcianoviereck commented on app.js:
```diff
@@ -0,0 +1,77 @@
+console.log(round)
+
+let arrays =
+console.log("I think variable arrays is related to animals");
```
But what do you think is inside the array? What is the value of the array?

Shvan1 replied:
Changed to: "I think the value will be undefined"

marcianoviereck commented on app.js:
```diff
@@ -0,0 +1,77 @@
+let myFavoriteAnimals = [cat, dog, sheep];
+console.log(myFavoriteAnimals);
+let favoriteAnimalOfDaan = "baby pig";
+const animals = "myFavoriteAnimals" + "" + "favoriteAnimalOfDaan";
```
Ah, this is not correct.

You're making a `string` with the value `myFavoriteAnimals` and then adding `favoriteAnimalOfDaan` to it.
So it will result in "myFavoriteAnimalsfavoriteAnimalOfDaan", instead of an `array` with the values in it.

Shvan1 replied:
Changed to: const animals = myFavoriteAnimals + "" + favoriteAnimalOfDaan;

marcianoviereck commented on app.js:
```diff
@@ -0,0 +1,77 @@
+
+let myString = "this is a test";
+console.log(myString);
+console.log(myString ["length"]);
```
myString.length is a better solution (I'm not even sure if `myString ["length"]` compiles)

Shvan1 replied:
Changed to: console.log(myString.length);

marcianoviereck commented on app.js:
```diff
@@ -0,0 +1,77 @@
+console.log(address);
+console.log(supermarkets);
+console.log(booleans);
+console.log(I think the type of num is number,
```
Did you forget tot add " " in the console log? (because I dont think this compiles)

Shvan1 replied:
Changed to: console.log("I think the type of num is number, type of address is string, type of supermarkets is array and type of booleans is boolean.");

marcianoviereck commented on app.js:
```diff
@@ -0,0 +1,77 @@
+var maxNumber = Math.pow(6/0);
+
+if (maxNumber === Infinity)
+console.log(0 / maxNumber);
```
I would recommend adding brackets here, so like this:
```
if (maxNumber === Infinity) {
  console.log(0 / maxNumber);
}
```

It makes it easier to avoid mistakes

Shvan1 replied:
Changed to: var maxNumber = Math.pow(6/0);

if (maxNumber === Infinity){
console.log(0 / maxNumber);
}

marcianoviereck reviewed (changes requested):
In general the homework looks good, but there are some remarks here and there.

And can you split up the homework in separate files?

> For all the following exercises create a new .js file. Try to find a proper name for each file or make a small comment about what it does inside for future reference.

marcianoviereck commented on app.js:
```diff
@@ -0,0 +1,77 @@
+
```
Can you split up the homework in separate files?

> For all the following exercises create a new .js file. Try to find a proper name for each file or make a small comment about what it does inside for future reference.

Shvan1 committed:
Update app.js

Shvan1 committed:
Add files via upload

Week3 Homeworkof JS

Shvan1 committed:
Add files via upload

Shvan1 committed:
Merge pull request #1 from Shvan1/JSweek3

JS Homework-week 3

wouterkleijn commented:
@marcianoviereck I noticed @Shvan1  made some edits. Is this okay now or does he need to make some further changes? Thanks!

marcianoviereck commented:
@wouterkleijn not all the feedback is resolved yet, as far as I can see in the 'files changed'. There are still some remarks of me which have not been fixed yet.

marcianoviereck reviewed (changes requested):
Not all my comments have been adressed yet, as far as I can see. Can you still make those changes?

Shvan1 committed:
Update app.js

I have split the questions and corrected some mistakes that figured out by Marciano.

marcianoviereck commented on app.js:
```diff
@@ -1,80 +1,142 @@
+//6.6
 let favoriteAnimalOfDaan = "baby pig";
-const animals = "myFavoriteAnimals" + "" + "favoriteAnimalOfDaan";
+const animals = myFavoriteAnimals + "" + favoriteAnimalOfDaan;
```
Ah this one is still incorrect,you should use `myFavoriteAnimals.push('baby pig')`. 
Read more about it here: https://developer.mozilla.org/nl/docs/Web/JavaScript/Reference/Global_Objects/Array/push

Shvan1 replied:
Changed to: const animals = myFavoriteAnimals.push('baby pig');
Just last week, we have covered the .push() method.

marcianoviereck reviewed (changes requested):
One last change request, for the rest the homework looks good!
If you fix that last one, then the homework is approved.

Shvan1 committed:
Update app.js

question 6.6 is edited.

marcianoviereck reviewed (approved):
Ah nice, thanks for fixing the homework! It's approved now :)
